apps/ensapi/src/handlers/resolution-api.ts

```diff
@@ -37,21 +30,36 @@ const getCanAccelerateResolution = simpleMemoized(canAccelerateResolution, 30_00
  */
 app.get(
   "/records/:name",
-  validate("param", routes.records.params),
-  validate("query", routes.records.query),
+  validate("param", z.object({ name: params.name })),
+  validate(
+    "query",
+    z
+      .object({
+        ...params.selectionParams.shape,
+        trace: params.trace,
+        accelerate: params.accelerate,
+      })
+      .transform((value) => {
+        const { trace, accelerate, ...selectionParams } = value;
+        const selection = params.selection.parse(selectionParams);
+        return { selection, trace, accelerate };
+      }),
+  ),
   async (c) => {
     const { name } = c.req.valid("param");
-    const { selection, trace: showTrace, accelerate: _accelerate } = c.req.valid("query");
-    const accelerate = _accelerate && getCanAccelerateResolution();
+    const { selection, trace: showTrace, accelerate } = c.req.valid("query");
+    const canAccelerate = c.var.canAccelerate;
 
     try {
       const { result, trace } = await captureTrace(() =>
-        resolveForward(name, selection, { accelerate }),
+        resolveForward(name, selection, { accelerate, canAccelerate }),
       );
 
       const response = {
         records: result,
-        accelerationAttempted: accelerate,
+
+        accelerationRequested: accelerate,
```

Comment:
i added this acceleration requested and acceleration attempted metadata to the resolution api, we can integrate it into ensadmin's tracing page later (in followup pr)